src/tests/ftest/util/dfuse_utils.py

```diff
@@ -173,21 +199,35 @@ def stop(self):
         """Stop dfuse
         Raises:
             CommandFailure: In case dfuse stop fails
+
+        Try to stop dfuse.  Try once nicely by using fusermount, then if that
+        fails try to pkill it to see if that works.  Abort based on the result
+        of the fusermount, as if pkill is necessary then dfuse itself has
+        not worked correctly.
+
+        Finally, try and remove the mount point, and that itself should work.
         """
+        self.log.info('Stopping dfuse at %s', self.mount_dir.value)
 
+        if self.mount_dir.value is None:
+            return
         cmd = "if [ -x '$(command -v fusermount)' ]; "
         cmd += "then fusermount -u {0}; else fusermount3 -u {0}; fi".\
                format(self.mount_dir.value)
         ret_code = general_utils.pcmd(self.hosts, cmd, timeout=30)
-        self.remove_mount_point()
-        if 0 not in ret_code:
+        if len(ret_code) > 1 or 0 not in ret_code:
             error_hosts = NodeSet(
                 ",".join(
                     [str(node_set) for code, node_set in ret_code.items()
                      if code != 0]))
+
+            cmd = "pkill dfuse --signal KILL"
```

Comment:
Is `pkill`ing `dfuse` as the non-root user expected to be permissible or would root generally be needed?

Reply:
pkill should work, but we probably also want a "sudo umount -f" afterwards, possibly twice.